ports/zephyr/main.c

```diff
@@ -104,8 +137,13 @@ int real_main(void) {
     mp_obj_list_append(mp_sys_path, MP_OBJ_NEW_QSTR(MP_QSTR_)); // current dir (or base dir of the script)
     mp_obj_list_init(mp_sys_argv, 0);
 
-    #if MICROPY_MODULE_FROZEN
-    pyexec_frozen_module("main.py");
+    #ifdef CONFIG_USB
+    usb_enable(NULL);
+    #endif
+
+    #if MICROPY_VFS
+    vfs_init();
+    pyexec_file_if_exists("main.py");
```

Comment:
The existing behaviour did support running main.py from frozen code (eg create a directory called `scripts/` and put `main.py` there).  So I think it's a good idea to keep that behaviour.

`pyexec_file_if_exists()` will actually work just like `pyexec_frozen_module()` if there's no VFS, so I'd suggest doing this:
```c
#if MICROPY_VFS
vfs_init();
#endif 

#if MICROPY_MODULE_FROZEN || MICROPY_VFS
pyexec_file_if_exists("main.py";)
#endif 
```

Reply:
Works for me, done.